src/main/java/org/opensearch/knn/plugin/rest/RestSearchModelHandler.java

```diff
@@ -69,9 +72,30 @@ private void checkUnSupportedParamsExists(RestRequest request) {
         throw new IllegalArgumentException(errorMessage);
     }
 
+    private void validateSizeParameter(RestRequest request) {
+        if (!request.hasParam(PARAM_SIZE)) {
+            return;
+        }
+        if (isSearchSizeValueValid(request.paramAsInt(PARAM_SIZE, 1))) {
+            return;
+        }
+        throw new IllegalArgumentException(
+            String.format("%s must be between %d and %d inclusive", PARAM_SIZE, SEARCH_MODEL_MIN_SIZE, SEARCH_MODEL_MAX_SIZE)
```

Comment:
Is it clear from the message that invalid value is for 'size parameter' param? If it's not obvious from scenario I think we need to add param name to the error message.

Reply:
Discussed offline, message has the param name, closing this